invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -113,19 +113,18 @@ def view(**kwargs):
             pid_value = kwargs.get("pid_value")
             is_preview = kwargs.get("is_preview")
 
-            def get_record():
+            def _get_record(read_function):
                 """Retrieve record."""
-                return service().read(id_=pid_value, identity=g.identity, expand=expand)
+                return read_function(id_=pid_value, identity=g.identity, expand=expand)
 
             if is_preview:
                 try:
-                    record = service().read_draft(
-                        id_=pid_value, identity=g.identity, expand=expand
-                    )
+                    record = _get_record(service().read_draft)
                 except NoResultFound:
-                    record = get_record()
+                    record = _get_record(service().read)
             else:
-                record = get_record()
+                record = _get_record(service().read)
```

Comment:
My "readability"/"dryness" take here would be to ask what these functions are trying to abstract or reduce duplication of. The answer is the passing of a same set of arguments, i.e. not a specific function. So something like: 

```python
read_kwargs = dict(id_=pid_value, identity=g.identity, expand=expand)
service = service()

if is_preview:
    try:
        record = service.read_draft(**read_kwargs)
    except NoResultFound:
        record = service.read(**read_kwargs)
else:
    record = service.read(**read_kwargs)
```

seems more straightforward / true (passing functions around is probably harder to follow / any function could have been passed to `_get_record` not just a function to get a record ...).

The same pattern can be reused for the other functions.

Reply:
Actually yes, _much_ better :smile: